src/odemis/gui/cont/tabs/fastem_project_ribbons_tab.py

```diff
@@ -0,0 +1,277 @@
+        sizey = units.readable_str(sizey, unit="m", sig=3)
+        self.data[RibbonColumnNames.POSX.value] = posx
+        self.data[RibbonColumnNames.POSY.value] = posy
+        self.data[RibbonColumnNames.SIZEX.value] = sizex
+        self.data[RibbonColumnNames.SIZEY.value] = sizey
+        self.data[RibbonColumnNames.ROT.value] = int(
+            math.degrees(self.roa.shape.rotation)
+        )
+
+    def on_cell_changing(self, new_value, row, col, grid):
+        """
+        Handles changes in cell values and updates associated data and ROA.
+
+        :param new_value: (str) The new value to set in the cell.
+        :param row: (int) The index of the row being modified.
+        :param col: (int) The index of the column being modified.
+        :param grid: (GridBase) The grid instance that contains the row.
+        :return: (str) The value to set in the cell.
+        """
+        value_to_set = new_value
+        rows = grid.rows
+        col = grid.columns[col]
+        if col.label == RibbonColumnNames.NAME.value:
+            value_to_set = make_compliant_string(value_to_set)
+            if not value_to_set:
+                value_to_set = "Ribbon"
+            current_slice_idx = self.data[RibbonColumnNames.SLICE_IDX.value]
+            if not RibbonRow.is_unique_name_slice_idx(
+                value_to_set, current_slice_idx, rows
+            ):
+                new_slice_idx = RibbonRow.find_next_slice_index(value_to_set, rows)
+                self.data[RibbonColumnNames.SLICE_IDX.value] = new_slice_idx
+                self.roa.slice_index.value = new_slice_idx
+                slice_index_col = grid.get_column_by_label(
+                    RibbonColumnNames.SLICE_IDX.value
+                )
+                grid.SetCellValue(row, slice_index_col.index, str(new_slice_idx))
+            self.data[RibbonColumnNames.NAME.value] = value_to_set
+            self.roa.shape.name.value = (
+                f"{value_to_set}_{self.data[RibbonColumnNames.SLICE_IDX.value]}"
+            )
+            self.roa.name.value = value_to_set
+        elif col.label == RibbonColumnNames.SLICE_IDX.value:
+            current_name = self.data[RibbonColumnNames.NAME.value]
+            if not RibbonRow.is_unique_name_slice_idx(
+                current_name, int(value_to_set), rows
+            ):
+                new_slice_idx = RibbonRow.find_next_slice_index(current_name, rows)
+                value_to_set = str(new_slice_idx)
+            self.data[RibbonColumnNames.SLICE_IDX.value] = int(value_to_set)
+            self.roa.slice_index.value = int(value_to_set)
+            self.roa.shape.name.value = (
+                f"{self.data[RibbonColumnNames.NAME.value]}_{value_to_set}"
+            )
+        elif col.label == RibbonColumnNames.POSX.value:
+            posy = self.data[RibbonColumnNames.POSY.value]
+            self.data[RibbonColumnNames.POSX.value] = posx = float(value_to_set)
```

Comment:
Can you add a comment explaining how incorrect strings are handled? Does it raise exception (in which case you should just add that info to the docstring)? Or is the "value_to_set" always of the right format (in which case you should indicate in the docstring)?